### ISSUE TITLE
Environment variables are not accessible in pipeline_config.groovy file

### ISSUE THREAD

**Issue by benjcleveland** [jenkinsci/templating-engine-plugin#217]
### Environment Information

- [x ] Jenkins version 2.289.1-lts
- [x ] Plugin version 2.2.2
- [ x] OS CentOS 7

### Description

We recently upgraded to JTE 2.2.2 and started getting the following error in one of our pipelines ```groovy.lang.MissingPropertyException: No such property: BRANCH_NAME for class: org.boozallen.plugins.jte.init.governance.config.dsl.PipelineConfigurationBuilder$BuilderMethod```. Previously we were running 2.0.4 and this was working, the pipeline_config.groovy file is trying to access the `env. BRANCH_NAME.` This only seems to fail when the pipeline configuration is supplied through the 'console' in  multibranch pipeline job.  This does seem to work correctly when the `pipeline_config.groovy` file is supplied through the scm option.

1. Create a new multibranch pipeline job.
2. In the 'Jenkins Template Engine' section select the option to provide a pipeline configuration
3. In the pipeline configuration, add the following 
```
application_environment {
  if (env.BRANCH_NAME == 'test') {    
  }
}
```
4. Run the job - this may need to be done twice, the first time will require approval of this signature `method groovy.lang.GroovyObject getProperty java.lang.String`

**Comment by steven-terrana** [comment]
hey @benjcleveland  - thanks for reporting! 

This is already fixed as part of #211. 

I'll get a 2.3 release out some time today.

**Comment by benjcleveland** [comment]
Awesome, thanks. We were able to work around the problem by pulling the environment variable out of the `application_environment` definition. We will try reverting back once we upgrade to 2.3.
